bundles/org.openhab.binding.warmup/src/main/java/org/openhab/binding/warmup/internal/handler/MyWarmupAccountHandler.java

```diff
@@ -0,0 +1,131 @@
+import org.openhab.binding.warmup.internal.model.query.QueryResponseDTO;
+
+/**
+ * @author James Melville - Initial contribution
+ */
+@NonNullByDefault
+public class MyWarmupAccountHandler extends BaseBridgeHandler {
+
+    private final MyWarmupApi api;
+
+    private @Nullable QueryResponseDTO queryResponse = null;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @Nullable WarmupDiscoveryService discoveryService;
+
+    public MyWarmupAccountHandler(Bridge thing, final HttpClient httpClient) {
+        super(thing);
+        api = new MyWarmupApi(httpClient, getConfigAs(MyWarmupConfigurationDTO.class));
+    }
+
+    @Override
+    public void initialize() {
+        MyWarmupConfigurationDTO config = getConfigAs(MyWarmupConfigurationDTO.class);
+        if (config.refreshIntervalSec >= 10) {
+            api.setConfiguration(config);
+            refreshJob = scheduler.scheduleWithFixedDelay(this::refreshFromServer, 0, config.refreshIntervalSec,
+                    TimeUnit.SECONDS);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Refresh interval misconfigured (minimum 10s)");
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void dispose() {
+        cancelRefresh();
+    }
+
+    public void cancelRefresh() {
+        if (refreshJob != null) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    public synchronized void refreshFromServer() {
+        try {
+            queryResponse = api.getStatus();
+        } catch (MyWarmupApiException e) {
+            queryResponse = null;
+            cancelRefresh();
+        } finally {
+            if (queryResponse != null) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Unable to contact MyWarmup");
+            }
+            refreshFromCache();
+        }
+    }
+
+    /**
+     * Trigger updates to all devices
+     */
+    public synchronized void refreshFromCache() {
+        notifyListeners(queryResponse);
+    }
+
+    public synchronized void scanDevices() {
+        if (discoveryService != null && queryResponse != null) {
+            discoveryService.refresh(queryResponse);
+        }
+    }
+
+    /**
+     * Initiate discovery
+     */
+    public void setDiscoveryService(final WarmupDiscoveryService discoveryService) {
+        this.discoveryService = discoveryService;
+    }
+
+    /**
+     *
+     * @return reference to the bridge's API
+     */
+    public MyWarmupApi getApi() {
+        return api;
+    }
+
+    private void notifyListeners(@Nullable QueryResponseDTO domain) {
+        if (discoveryService != null && queryResponse != null) {
+            discoveryService.refresh(queryResponse);
+        }
```

Comment:
A discovery should only be triggered from the discovery service itself. So this part might not be a good idea. What is the reason to call it here?